editor/src/clj/dynamo/graph.clj

```diff
@@ -237,15 +238,35 @@
 ;; ---------------------------------------------------------------------------
 ;; Using transaction data
 ;; ---------------------------------------------------------------------------
-(defn tx-data-nodes-added
-  "Returns a list of the node-ids added given a list of transaction steps, (tx-data)."
+
+(defn tx-data-step-types
+  "Given a sequence of transaction steps, returns a sequence of keywords
+  identifying the type of each pending transaction step. Used in tests."
+  [txs]
+  (map it/tx-step-type
+       (flatten txs)))
```

Comment:
Hmm, `flatten` was there before, but it's actually concerning. Transactions explicitly support nils (`apply-tx` skips them), but `flatten` preserves nils. Maybe use `coll/flatten-xf` and transducers for better semantics (`flatten-xf` skips nils) and performance?